<?xml version='1.0' encoding='UTF-8'?>
<rss xmlns:atom="http://www.w3.org/2005/Atom" xmlns:content="http://purl.org/rss/1.0/modules/content/" version="2.0">
  <channel>
    <title>CVE Feed for tensorflow -- all</title>
    <link>https://raw.githubusercontent.com/deepseas/cvelistV5monitor/main/feeds/tensorflow/all.rss</link>
    <description>The latest CVEs for tensorflow -- all products</description>
    <docs>http://www.rssboard.org/rss-specification</docs>
    <generator>python-feedgen</generator>
    <lastBuildDate>Sat, 03 Aug 2024 11:20:05 +0000</lastBuildDate>
    <ttl>60</ttl>
    <item>
      <title>CVE-2022-35959|2024-08-03T09:51:59.851Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35959</link>
      <description>TensorFlow is an open source platform for machine learning. The implementation of `AvgPool3DGradOp` does not fully validate the input `orig_input_shape`. This results in an overflow that results in a `CHECK` failure which can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 9178ac9d6389bdc54638ab913ea0e419234d14eb. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35959|2024-08-03T09:51:59.851Z</guid>
      <pubDate>Fri, 16 Sep 2022 19:55:11 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36005|2024-08-03T09:51:59.861Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36005</link>
      <description>TensorFlow is an open source platform for machine learning. When `tf.quantization.fake_quant_with_min_max_vars_gradient` receives input `min` or `max` that is nonscalar, it gives a `CHECK` fail that can trigger a denial of service attack. We have patched the issue in GitHub commit f3cf67ac5705f4f04721d15e485e192bb319feed. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36005|2024-08-03T09:51:59.861Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:10:31 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29199|2024-08-03T06:17:54.045Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29199</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.LoadAndRemapMatrix does not fully validate the input arguments. This results in a `CHECK`-failure which can be used to trigger a denial of service attack. The code assumes `initializing_values` is a vector but there is no validation for this before accessing its value. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29199|2024-08-03T06:17:54.045Z</guid>
      <pubDate>Fri, 20 May 2022 21:40:12 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36014|2024-08-03T09:51:59.971Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36014</link>
      <description>TensorFlow is an open source platform for machine learning. When `mlir::tfg::TFOp::nameAttr` receives null type list attributes, it crashes. We have patched the issue in GitHub commits 3a754740d5414e362512ee981eefba41561a63a6 and a0f0b9a21c9270930457095092f558fbad4c03e5. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36014|2024-08-03T09:51:59.971Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:35:11 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29210|2024-08-03T06:17:54.137Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29210</link>
      <description>TensorFlow is an open source platform for machine learning. In version 2.8.0, the `TensorKey` hash function used total estimated `AllocatedBytes()`, which (a) is an estimate per tensor, and (b) is a very poor hash function for constants (e.g. `int32_t`). It also tried to access individual tensor bytes through `tensor.data()` of size `AllocatedBytes()`. This led to ASAN failures because the `AllocatedBytes()` is an estimate of total bytes allocated by a tensor, including any pointed-to constructs (e.g. strings), and does not refer to contiguous bytes in the `.data()` buffer. The discoverers could not use this byte vector anyway because types such as `tstring` include pointers, whereas they needed to hash the string values themselves. This issue is patched in Tensorflow versions 2.9.0 and 2.8.1.</description>
      <guid isPermaLink="false">CVE-2022-29210|2024-08-03T06:17:54.137Z</guid>
      <pubDate>Fri, 20 May 2022 23:25:21 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35963|2024-08-03T09:51:59.270Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35963</link>
      <description>TensorFlow is an open source platform for machine learning. The implementation of `FractionalAvgPoolGrad` does not fully validate the input `orig_input_tensor_shape`. This results in an overflow that results in a `CHECK` failure which can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 03a659d7be9a1154fdf5eeac221e5950fec07dad. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35963|2024-08-03T09:51:59.270Z</guid>
      <pubDate>Fri, 16 Sep 2022 20:10:10 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29207|2024-08-03T06:17:54.201Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29207</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, multiple TensorFlow operations misbehave in eager mode when the resource handle provided to them is invalid. In graph mode, it would have been impossible to perform these API calls, but migration to TF 2.x eager mode opened up this vulnerability. If the resource handle is empty, then a reference is bound to a null pointer inside TensorFlow codebase (various codepaths). This is undefined behavior. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29207|2024-08-03T06:17:54.201Z</guid>
      <pubDate>Fri, 20 May 2022 22:10:12 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29200|2024-08-03T06:17:54.123Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29200</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.LSTMBlockCell` does not fully validate the input arguments. This results in a `CHECK`-failure which can be used to trigger a denial of service attack. The code does not validate the ranks of any of the arguments to this API call. This results in `CHECK`-failures when the elements of the tensor are accessed. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29200|2024-08-03T06:17:54.123Z</guid>
      <pubDate>Fri, 20 May 2022 21:30:14 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23585|2024-08-03T03:43:46.959Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23585</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. When decoding PNG images TensorFlow can produce a memory leak if the image is invalid. After calling `png::CommonInitDecode(..., &amp;decode)`, the `decode` value contains allocated buffers which can only be freed by calling `png::CommonFreeDecode(&amp;decode)`. However, several error case in the function implementation invoke the `OP_REQUIRES` macro which immediately terminates the execution of the function, without allowing for the memory free to occur. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23585|2024-08-03T03:43:46.959Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:27 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23577|2024-08-03T03:43:46.849Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23577</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. The implementation of `GetInitOp` is vulnerable to a crash caused by dereferencing a null pointer. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23577|2024-08-03T03:43:46.849Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:31 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36002|2024-08-03T09:51:59.529Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36002</link>
      <description>TensorFlow is an open source platform for machine learning. When `Unbatch` receives a nonscalar input `id`, it gives a `CHECK` fail that can trigger a denial of service attack. We have patched the issue in GitHub commit 4419d10d576adefa36b0e0a9425d2569f7c0189f. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36002|2024-08-03T09:51:59.529Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:10:15 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23583|2024-08-03T03:43:46.960Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23583</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. A malicious user can cause a denial of service by altering a `SavedModel` such that any binary op would trigger `CHECK` failures. This occurs when the protobuf part corresponding to the tensor arguments is modified such that the `dtype` no longer matches the `dtype` expected by the op. In that case, calling the templated binary operator for the binary op would receive corrupted data, due to the type confusion involved. If `Tin` and `Tout` don't match the type of data in `out` and `input_*` tensors then `flat&lt;*&gt;` would interpret it wrongly. In most cases, this would be a silent failure, but we have noticed scenarios where this results in a `CHECK` crash, hence a denial of service. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23583|2024-08-03T03:43:46.960Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:18 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29209|2024-08-03T06:17:54.118Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29209</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the macros that TensorFlow uses for writing assertions (e.g., `CHECK_LT`, `CHECK_GT`, etc.) have an incorrect logic when comparing `size_t` and `int` values. Due to type conversion rules, several of the macros would trigger incorrectly. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29209|2024-08-03T06:17:54.118Z</guid>
      <pubDate>Fri, 20 May 2022 23:25:14 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23580|2024-08-03T03:43:46.550Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23580</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. During shape inference, TensorFlow can allocate a large vector based on a value from a tensor controlled by the user. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23580|2024-08-03T03:43:46.550Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:25 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35970|2024-08-03T09:51:59.226Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35970</link>
      <description>TensorFlow is an open source platform for machine learning. If `QuantizedInstanceNorm` is given `x_min` or `x_max` tensors of a nonzero rank, it results in a segfault that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 785d67a78a1d533759fcd2f5e8d6ef778de849e0. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35970|2024-08-03T09:51:59.226Z</guid>
      <pubDate>Fri, 16 Sep 2022 20:45:13 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23579|2024-08-03T03:43:46.969Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23579</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. The Grappler optimizer in TensorFlow can be used to cause a denial of service by altering a `SavedModel` such that `SafeToRemoveIdentity` would trigger `CHECK` failures. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23579|2024-08-03T03:43:46.969Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:26 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36026|2024-08-03T09:51:59.926Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36026</link>
      <description>TensorFlow is an open source platform for machine learning. If `QuantizeAndDequantizeV3` is given a nonscalar `num_bits` input tensor, it results in a `CHECK` fail that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit f3f9cb38ecfe5a8a703f2c4a8fead434ef291713. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36026|2024-08-03T09:51:59.926Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:05:20 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23565|2024-08-03T03:43:46.543Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23565</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. An attacker can trigger denial of service via assertion failure by altering a `SavedModel` on disk such that `AttrDef`s of some operation are duplicated. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23565|2024-08-03T03:43:46.543Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:40 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23582|2024-08-03T03:43:46.960Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23582</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. A malicious user can cause a denial of service by altering a `SavedModel` such that `TensorByteSize` would trigger `CHECK` failures. `TensorShape` constructor throws a `CHECK`-fail if shape is partial or has a number of elements that would overflow the size of an `int`. The `PartialTensorShape` constructor instead does not cause a `CHECK`-abort if the shape is partial, which is exactly what this function needs to be able to return `-1`. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23582|2024-08-03T03:43:46.960Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:17 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35999|2024-08-03T09:51:59.743Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35999</link>
      <description>TensorFlow is an open source platform for machine learning. When `Conv2DBackpropInput` receives empty `out_backprop` inputs (e.g. `[3, 1, 0, 1]`), the current CPU/GPU kernels `CHECK` fail (one with dnnl, the other with cudnn). This can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 27a65a43cf763897fecfa5cdb5cc653fc5dd0346. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35999|2024-08-03T09:51:59.743Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:15:28 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35994|2024-08-03T09:51:59.941Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35994</link>
      <description>TensorFlow is an open source platform for machine learning. When `CollectiveGather` receives an scalar input `input`, it gives a `CHECK` fails that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit c1f491817dec39a26be3c574e86a88c30f3c4770. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35994|2024-08-03T09:51:59.941Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:20:31 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35990|2024-08-03T09:51:59.863Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35990</link>
      <description>TensorFlow is an open source platform for machine learning. When `tf.quantization.fake_quant_with_min_max_vars_per_channel_gradient` receives input `min` or `max` of rank other than 1, it gives a `CHECK` fail that can trigger a denial of service attack. We have patched the issue in GitHub commit f3cf67ac5705f4f04721d15e485e192bb319feed. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range.There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35990|2024-08-03T09:51:59.863Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:00:12 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36016|2024-08-03T09:51:59.918Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36016</link>
      <description>TensorFlow is an open source platform for machine learning. When `tensorflow::full_type::SubstituteFromAttrs` receives a `FullTypeDef&amp; t` that is not exactly three args, it triggers a `CHECK`-fail instead of returning a status. We have patched the issue in GitHub commit 6104f0d4091c260ce9352f9155f7e9b725eab012. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36016|2024-08-03T09:51:59.918Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:10:36 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35939|2024-08-03T09:51:59.104Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35939</link>
      <description>TensorFlow is an open source platform for machine learning. The `ScatterNd` function takes an input argument that determines the indices of of the output tensor. An input index greater than the output tensor or less than zero will either write content at the wrong index or trigger a crash. We have patched the issue in GitHub commit b4d4b4cb019bd7240a52daa4ba61e3cc814f0384. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35939|2024-08-03T09:51:59.104Z</guid>
      <pubDate>Fri, 16 Sep 2022 19:40:11 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23563|2024-08-03T03:43:46.464Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23563</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. In multiple places, TensorFlow uses `tempfile.mktemp` to create temporary files. While this is acceptable in testing, in utilities and libraries it is dangerous as a different process can create the file between the check for the filename in `mktemp` and the actual creation of the file by a subsequent operation (a TOC/TOU type of weakness). In several instances, TensorFlow was supposed to actually create a temporary directory instead of a file. This logic bug is hidden away by the `mktemp` function usage. We have patched the issue in several commits, replacing `mktemp` with the safer `mkstemp`/`mkdtemp` functions, according to the usage pattern. Users are advised to upgrade as soon as possible.</description>
      <guid isPermaLink="false">CVE-2022-23563|2024-08-03T03:43:46.464Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:38 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36004|2024-08-03T09:51:59.678Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36004</link>
      <description>TensorFlow is an open source platform for machine learning. When `tf.random.gamma` receives large input shape and rates, it gives a `CHECK` fail that can trigger a denial of service attack. We have patched the issue in GitHub commit 552bfced6ce4809db5f3ca305f60ff80dd40c5a3. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36004|2024-08-03T09:51:59.678Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:10:26 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29208|2024-08-03T06:17:54.350Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29208</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.EditDistance` has incomplete validation. Users can pass negative values to cause a segmentation fault based denial of service. In multiple places throughout the code, one may compute an index for a write operation. However, the existing validation only checks against the upper bound of the array. Hence, it is possible to write before the array by massaging the input to generate negative values for `loc`. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29208|2024-08-03T06:17:54.350Z</guid>
      <pubDate>Fri, 20 May 2022 22:30:13 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35991|2024-08-03T09:51:59.552Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35991</link>
      <description>TensorFlow is an open source platform for machine learning. When `TensorListScatter` and `TensorListScatterV2` receive an `element_shape` of a rank greater than one, they give a `CHECK` fail that can trigger a denial of service attack. We have patched the issue in GitHub commit bb03fdf4aae944ab2e4b35c7daa051068a8b7f61. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35991|2024-08-03T09:51:59.552Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:20:16 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35966|2024-08-03T09:51:59.499Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35966</link>
      <description>TensorFlow is an open source platform for machine learning. If `QuantizedAvgPool` is given `min_input` or `max_input` tensors of a nonzero rank, it results in a segfault that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 7cdf9d4d2083b739ec81cfdace546b0c99f50622. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35966|2024-08-03T09:51:59.499Z</guid>
      <pubDate>Fri, 16 Sep 2022 20:35:15 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35988|2024-08-03T09:51:59.671Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35988</link>
      <description>TensorFlow is an open source platform for machine learning. When `tf.linalg.matrix_rank` receives an empty input `a`, the GPU kernel gives a `CHECK` fail that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit c55b476aa0e0bd4ee99d0f3ad18d9d706cd1260a. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35988|2024-08-03T09:51:59.671Z</guid>
      <pubDate>Fri, 16 Sep 2022 21:35:10 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23581|2024-08-03T03:43:46.546Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23581</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. The Grappler optimizer in TensorFlow can be used to cause a denial of service by altering a `SavedModel` such that `IsSimplifiableReshape` would trigger `CHECK` failures. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23581|2024-08-03T03:43:46.546Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:24 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36013|2024-08-03T09:51:59.650Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36013</link>
      <description>TensorFlow is an open source platform for machine learning. When `mlir::tfg::GraphDefImporter::ConvertNodeDef` tries to convert NodeDefs without an op name, it crashes. We have patched the issue in GitHub commit a0f0b9a21c9270930457095092f558fbad4c03e5. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36013|2024-08-03T09:51:59.650Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:30:14 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29193|2024-08-03T06:17:54.253Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29193</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.TensorSummaryV2` does not fully validate the input arguments. This results in a `CHECK`-failure which can be used to trigger a denial of service attack. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29193|2024-08-03T06:17:54.253Z</guid>
      <pubDate>Fri, 20 May 2022 21:20:14 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36015|2024-08-03T09:51:59.737Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36015</link>
      <description>TensorFlow is an open source platform for machine learning. When `RangeSize` receives values that do not fit into an `int64_t`, it crashes. We have patched the issue in GitHub commit 37e64539cd29fcfb814c4451152a60f5d107b0f0. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36015|2024-08-03T09:51:59.737Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:55:21 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35971|2024-08-03T09:51:59.270Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35971</link>
      <description>TensorFlow is an open source platform for machine learning. If `FakeQuantWithMinMaxVars` is given `min` or `max` tensors of a nonzero rank, it results in a `CHECK` fail that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 785d67a78a1d533759fcd2f5e8d6ef778de849e0. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35971|2024-08-03T09:51:59.270Z</guid>
      <pubDate>Fri, 16 Sep 2022 20:50:10 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35952|2024-08-03T09:51:59.175Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35952</link>
      <description>TensorFlow is an open source platform for machine learning. The `UnbatchGradOp` function takes an argument `id` that is assumed to be a scalar. A nonscalar `id` can trigger a `CHECK` failure and crash the program. It also requires its argument `batch_index` to contain three times the number of elements as indicated in its `batch_index.dim_size(0)`. An incorrect `batch_index` can trigger a `CHECK` failure and crash the program. We have patched the issue in GitHub commit 5f945fc6409a3c1e90d6970c9292f805f6e6ddf2. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35952|2024-08-03T09:51:59.175Z</guid>
      <pubDate>Fri, 16 Sep 2022 19:50:10 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35986|2024-08-03T09:51:59.809Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35986</link>
      <description>TensorFlow is an open source platform for machine learning. If `RaggedBincount` is given an empty input tensor `splits`, it results in a segfault that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 7a4591fd4f065f4fa903593bc39b2f79530a74b8. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35986|2024-08-03T09:51:59.809Z</guid>
      <pubDate>Fri, 16 Sep 2022 21:45:13 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29192|2024-08-03T06:17:53.991Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29192</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.QuantizeAndDequantizeV4Grad` does not fully validate the input arguments. This results in a `CHECK`-failure which can be used to trigger a denial of service attack. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29192|2024-08-03T06:17:53.991Z</guid>
      <pubDate>Fri, 20 May 2022 20:30:14 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35979|2024-08-03T09:51:59.270Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35979</link>
      <description>TensorFlow is an open source platform for machine learning. If `QuantizedRelu` or `QuantizedRelu6` are given nonscalar inputs for `min_features` or `max_features`, it results in a segfault that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 49b3824d83af706df0ad07e4e677d88659756d89. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35979|2024-08-03T09:51:59.270Z</guid>
      <pubDate>Fri, 16 Sep 2022 21:10:10 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29216|2024-08-03T06:17:54.261Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29216</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, TensorFlow's `saved_model_cli` tool is vulnerable to a code injection. This can be used to open a reverse shell. This code path was maintained for compatibility reasons as the maintainers had several test cases where numpy expressions were used as arguments. However, given that the tool is always run manually, the impact of this is still not severe. The maintainers have now removed the `safe=False` argument, so all parsing is done without calling `eval`. The patch is available in versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4.</description>
      <guid isPermaLink="false">CVE-2022-29216|2024-08-03T06:17:54.261Z</guid>
      <pubDate>Fri, 20 May 2022 23:35:13 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35941|2024-08-03T09:51:59.193Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35941</link>
      <description>TensorFlow is an open source platform for machine learning. The `AvgPoolOp` function takes an argument `ksize` that must be positive but is not checked. A negative `ksize` can trigger a `CHECK` failure and crash the program. We have patched the issue in GitHub commit 3a6ac52664c6c095aa2b114e742b0aa17fdce78f. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds to this issue.</description>
      <guid isPermaLink="false">CVE-2022-35941|2024-08-03T09:51:59.193Z</guid>
      <pubDate>Fri, 16 Sep 2022 19:45:14 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29197|2024-08-03T06:17:54.098Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29197</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.UnsortedSegmentJoin` does not fully validate the input arguments. This results in a `CHECK`-failure which can be used to trigger a denial of service attack. The code assumes `num_segments` is a scalar but there is no validation for this before accessing its value. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29197|2024-08-03T06:17:54.098Z</guid>
      <pubDate>Fri, 20 May 2022 21:55:18 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35960|2024-08-03T09:51:59.580Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35960</link>
      <description>TensorFlow is an open source platform for machine learning. In `core/kernels/list_kernels.cc's TensorListReserve`, `num_elements` is assumed to be a tensor of size 1. When a `num_elements` of more than 1 element is provided, then `tf.raw_ops.TensorListReserve` fails the `CHECK_EQ` in `CheckIsAlignedAndSingleElement`. We have patched the issue in GitHub commit b5f6fbfba76576202b72119897561e3bd4f179c7. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35960|2024-08-03T09:51:59.580Z</guid>
      <pubDate>Fri, 16 Sep 2022 20:00:15 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23584|2024-08-03T03:43:47.079Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23584</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. A malicious user can cause a use after free behavior when decoding PNG images. After `png::CommonFreeDecode(&amp;decode)` gets called, the values of `decode.width` and `decode.height` are in an unspecified state. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23584|2024-08-03T03:43:47.079Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:16 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29201|2024-08-03T06:17:54.111Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29201</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.QuantizedConv2D` does not fully validate the input arguments. In this case, references get bound to `nullptr` for each argument that is empty. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29201|2024-08-03T06:17:54.111Z</guid>
      <pubDate>Fri, 20 May 2022 23:00:15 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36001|2024-08-03T09:51:59.864Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36001</link>
      <description>TensorFlow is an open source platform for machine learning. When `DrawBoundingBoxes` receives an input `boxes` that is not of dtype `float`, it gives a `CHECK` fail that can trigger a denial of service attack. We have patched the issue in GitHub commit da0d65cdc1270038e72157ba35bf74b85d9bda11. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36001|2024-08-03T09:51:59.864Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:10:10 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35974|2024-08-03T09:51:59.855Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35974</link>
      <description>TensorFlow is an open source platform for machine learning. If `QuantizeDownAndShrinkRange` is given nonscalar inputs for `input_min` or `input_max`, it results in a segfault that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 73ad1815ebcfeb7c051f9c2f7ab5024380ca8613. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35974|2024-08-03T09:51:59.855Z</guid>
      <pubDate>Fri, 16 Sep 2022 21:05:12 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29212|2024-08-03T06:17:54.081Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29212</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, certain TFLite models that were created using TFLite model converter would crash when loaded in the TFLite interpreter. The culprit is that during quantization the scale of values could be greater than 1 but code was always assuming sub-unit scaling. Thus, since code was calling `QuantizeMultiplierSmallerThanOneExp`, the `TFLITE_CHECK_LT` assertion would trigger and abort the process. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29212|2024-08-03T06:17:54.081Z</guid>
      <pubDate>Fri, 20 May 2022 23:15:15 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29191|2024-08-03T06:17:54.087Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29191</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.GetSessionTensor` does not fully validate the input arguments. This results in a `CHECK`-failure which can be used to trigger a denial of service attack. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29191|2024-08-03T06:17:54.087Z</guid>
      <pubDate>Fri, 20 May 2022 20:50:11 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35992|2024-08-03T09:51:59.517Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35992</link>
      <description>TensorFlow is an open source platform for machine learning. When `TensorListFromTensor` receives an `element_shape` of a rank greater than one, it gives a `CHECK` fail that can trigger a denial of service attack. We have patched the issue in GitHub commit 3db59a042a38f4338aa207922fa2f476e000a6ee. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35992|2024-08-03T09:51:59.517Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:20:21 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29195|2024-08-03T06:17:54.372Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29195</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.StagePeek` does not fully validate the input arguments. This results in a `CHECK`-failure which can be used to trigger a denial of service attack. The code assumes `index` is a scalar but there is no validation for this before accessing its value. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29195|2024-08-03T06:17:54.372Z</guid>
      <pubDate>Fri, 20 May 2022 22:00:15 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35984|2024-08-03T09:51:59.627Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35984</link>
      <description>TensorFlow is an open source platform for machine learning. `ParameterizedTruncatedNormal` assumes `shape` is of type `int32`. A valid `shape` of type `int64` results in a mismatched type `CHECK` fail that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 72180be03447a10810edca700cbc9af690dfeb51. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35984|2024-08-03T09:51:59.627Z</guid>
      <pubDate>Fri, 16 Sep 2022 21:40:14 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35940|2024-08-03T09:51:59.069Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35940</link>
      <description>TensorFlow is an open source platform for machine learning. The `RaggedRangOp` function takes an argument `limits` that is eventually used to construct a `TensorShape` as an `int64`. If `limits` is a very large float, it can overflow when converted to an `int64`. This triggers an `InvalidArgument` but also throws an abort signal that crashes the program. We have patched the issue in GitHub commit 37cefa91bee4eace55715eeef43720b958a01192. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35940|2024-08-03T09:51:59.069Z</guid>
      <pubDate>Fri, 16 Sep 2022 19:45:20 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35937|2024-08-03T09:51:59.208Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35937</link>
      <description>TensorFlow is an open source platform for machine learning. The `GatherNd` function takes arguments that determine the sizes of inputs and outputs. If the inputs given are greater than or equal to the sizes of the outputs, an out-of-bounds memory read is triggered. This issue has been patched in GitHub commit 595a65a3e224a0362d7e68c2213acfc2b499a196. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35937|2024-08-03T09:51:59.208Z</guid>
      <pubDate>Fri, 16 Sep 2022 19:40:20 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29206|2024-08-03T06:17:54.022Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29206</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.SparseTensorDenseAdd` does not fully validate the input arguments. In this case, a reference gets bound to a `nullptr` during kernel execution. This is undefined behavior. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29206|2024-08-03T06:17:54.022Z</guid>
      <pubDate>Fri, 20 May 2022 22:15:13 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29196|2024-08-03T06:17:54.117Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29196</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.Conv3DBackpropFilterV2` does not fully validate the input arguments. This results in a `CHECK`-failure which can be used to trigger a denial of service attack. The code does not validate that the `filter_sizes` argument is a vector. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29196|2024-08-03T06:17:54.117Z</guid>
      <pubDate>Fri, 20 May 2022 21:55:12 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35997|2024-08-03T09:51:59.513Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35997</link>
      <description>TensorFlow is an open source platform for machine learning. If `tf.sparse.cross` receives an input `separator` that is not a scalar, it gives a `CHECK` fail that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 83dcb4dbfa094e33db084e97c4d0531a559e0ebf. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35997|2024-08-03T09:51:59.513Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:15:18 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23578|2024-08-03T03:43:46.557Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23578</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. If a graph node is invalid, TensorFlow can leak memory in the implementation of `ImmutableExecutorState::Initialize`. Here, we set `item-&gt;kernel` to `nullptr` but it is a simple `OpKernel*` pointer so the memory that was previously allocated to it would leak. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23578|2024-08-03T03:43:46.557Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:30 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35983|2024-08-03T09:51:59.844Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35983</link>
      <description>TensorFlow is an open source platform for machine learning. If `Save` or `SaveSlices` is run over tensors of an unsupported `dtype`, it results in a `CHECK` fail that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 5dd7b86b84a864b834c6fa3d7f9f51c87efa99d4. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35983|2024-08-03T09:51:59.844Z</guid>
      <pubDate>Fri, 16 Sep 2022 21:40:10 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29213|2024-08-03T06:17:54.052Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29213</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the `tf.compat.v1.signal.rfft2d` and `tf.compat.v1.signal.rfft3d` lack input validation and under certain condition can result in crashes (due to `CHECK`-failures). Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29213|2024-08-03T06:17:54.052Z</guid>
      <pubDate>Fri, 20 May 2022 23:30:15 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23560|2024-08-03T03:43:46.562Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23560</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. An attacker can craft a TFLite model that would allow limited reads and writes outside of arrays in TFLite. This exploits missing validation in the conversion from sparse tensors to dense tensors. The fix is included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range. Users are advised to upgrade as soon as possible.</description>
      <guid isPermaLink="false">CVE-2022-23560|2024-08-03T03:43:46.562Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:36 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35989|2024-08-03T09:51:59.696Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35989</link>
      <description>TensorFlow is an open source platform for machine learning. When `MaxPool` receives a window size input array `ksize` with dimensions greater than its input tensor `input`, the GPU kernel gives a `CHECK` fail that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 32d7bd3defd134f21a4e344c8dfd40099aaf6b18. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35989|2024-08-03T09:51:59.696Z</guid>
      <pubDate>Fri, 16 Sep 2022 21:35:15 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35998|2024-08-03T09:51:59.844Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35998</link>
      <description>TensorFlow is an open source platform for machine learning. If `EmptyTensorList` receives an input `element_shape` with more than one dimension, it gives a `CHECK` fail that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit c8ba76d48567aed347508e0552a257641931024d. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35998|2024-08-03T09:51:59.844Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:15:23 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35938|2024-08-03T09:51:59.145Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35938</link>
      <description>TensorFlow is an open source platform for machine learning. The `GatherNd` function takes arguments that determine the sizes of inputs and outputs. If the inputs given are greater than or equal to the sizes of the outputs, an out-of-bounds memory read or a crash is triggered. This issue has been patched in GitHub commit 4142e47e9e31db481781b955ed3ff807a781b494. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35938|2024-08-03T09:51:59.145Z</guid>
      <pubDate>Fri, 16 Sep 2022 19:35:10 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23557|2024-08-03T03:43:46.509Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23557</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. An attacker can craft a TFLite model that would trigger a division by zero in `BiasAndClamp` implementation. There is no check that the `bias_size` is non zero. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23557|2024-08-03T03:43:46.509Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:44 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35968|2024-08-03T09:51:59.379Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35968</link>
      <description>TensorFlow is an open source platform for machine learning. The implementation of `AvgPoolGrad` does not fully validate the input `orig_input_shape`. This results in a `CHECK` failure which can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 3a6ac52664c6c095aa2b114e742b0aa17fdce78f. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35968|2024-08-03T09:51:59.379Z</guid>
      <pubDate>Fri, 16 Sep 2022 20:40:10 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23575|2024-08-03T03:43:46.468Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23575</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. The implementation of `OpLevelCostEstimator::CalculateTensorSize` is vulnerable to an integer overflow if an attacker can create an operation which would involve a tensor with large enough number of elements. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23575|2024-08-03T03:43:46.468Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:23 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36017|2024-08-03T09:51:59.872Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36017</link>
      <description>TensorFlow is an open source platform for machine learning. If `Requantize` is given `input_min`, `input_max`, `requested_output_min`, `requested_output_max` tensors of a nonzero rank, it results in a segfault that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 785d67a78a1d533759fcd2f5e8d6ef778de849e0. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36017|2024-08-03T09:51:59.872Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:50:10 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23576|2024-08-03T03:43:46.574Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23576</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. The implementation of `OpLevelCostEstimator::CalculateOutputSize` is vulnerable to an integer overflow if an attacker can create an operation which would involve tensors with large enough number of elements. We can have a large enough number of dimensions in `output_shape.dim()` or just a small number of dimensions being large enough to cause an overflow in the multiplication. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23576|2024-08-03T03:43:46.574Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:22 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23572|2024-08-03T03:43:46.825Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23572</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. Under certain scenarios, TensorFlow can fail to specialize a type during shape inference. This case is covered by the `DCHECK` function however, `DCHECK` is a no-op in production builds and an assertion failure in debug builds. In the first case execution proceeds to the `ValueOrDie` line. This results in an assertion failure as `ret` contains an error `Status`, not a value. In the second case we also get a crash due to the assertion failure. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, and TensorFlow 2.6.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23572|2024-08-03T03:43:46.825Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:29 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36011|2024-08-03T09:51:59.601Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36011</link>
      <description>TensorFlow is an open source platform for machine learning. When `mlir::tfg::ConvertGenericFunctionToFunctionDef` is given empty function attributes, it gives a null dereference. We have patched the issue in GitHub commit 1cf45b831eeb0cab8655c9c7c5d06ec6f45fc41b. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36011|2024-08-03T09:51:59.601Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:30:21 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35996|2024-08-03T09:51:59.680Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35996</link>
      <description>TensorFlow is an open source platform for machine learning. If `Conv2D` is given empty `input` and the `filter` and `padding` sizes are valid, the output is all-zeros. This causes division-by-zero floating point exceptions that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 611d80db29dd7b0cfb755772c69d60ae5bca05f9. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35996|2024-08-03T09:51:59.680Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:55:10 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23590|2024-08-03T03:43:46.906Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23590</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. A `GraphDef` from a TensorFlow `SavedModel` can be maliciously altered to cause a TensorFlow process to crash due to encountering a `StatusOr` value that is an error and forcibly extracting the value from it. We have patched the issue in multiple GitHub commits and these will be included in TensorFlow 2.8.0 and TensorFlow 2.7.1, as both are affected.</description>
      <guid isPermaLink="false">CVE-2022-23590|2024-08-03T03:43:46.906Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:10 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23562|2024-08-03T03:43:46.582Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23562</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. The implementation of `Range` suffers from integer overflows. These can trigger undefined behavior or, in some scenarios, extremely large allocations. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23562|2024-08-03T03:43:46.582Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:39 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23589|2024-08-03T03:43:46.987Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23589</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. Under certain scenarios, Grappler component of TensorFlow can trigger a null pointer dereference. There are 2 places where this can occur, for the same malicious alteration of a `SavedModel` file (fixing the first one would trigger the same dereference in the second place). First, during constant folding, the `GraphDef` might not have the required nodes for the binary operation. If a node is missing, the correposning `mul_*child` would be null, and the dereference in the subsequent line would be incorrect. We have a similar issue during `IsIdentityConsumingSwitch`. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23589|2024-08-03T03:43:46.987Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:20 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35934|2024-08-03T09:51:59.038Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35934</link>
      <description>TensorFlow is an open source platform for machine learning. The implementation of tf.reshape op in TensorFlow is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by overflowing the number of elements in a tensor. This issue has been patched in GitHub commit 61f0f9b94df8c0411f0ad0ecc2fec2d3f3c33555. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35934|2024-08-03T09:51:59.038Z</guid>
      <pubDate>Fri, 16 Sep 2022 19:30:13 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36003|2024-08-03T09:51:59.531Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36003</link>
      <description>TensorFlow is an open source platform for machine learning. When `RandomPoissonV2` receives large input shape and rates, it gives a `CHECK` fail that can trigger a denial of service attack. We have patched the issue in GitHub commit 552bfced6ce4809db5f3ca305f60ff80dd40c5a3. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36003|2024-08-03T09:51:59.531Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:10:21 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23561|2024-08-03T03:43:46.500Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23561</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. An attacker can craft a TFLite model that would cause a write outside of bounds of an array in TFLite. In fact, the attacker can override the linked list used by the memory allocator. This can be leveraged for an arbitrary write primitive under certain conditions. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23561|2024-08-03T03:43:46.500Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:46 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23587|2024-08-03T03:43:46.906Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23587</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. Under certain scenarios, Grappler component of TensorFlow is vulnerable to an integer overflow during cost estimation for crop and resize. Since the cropping parameters are user controlled, a malicious person can trigger undefined behavior. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23587|2024-08-03T03:43:46.906Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:14 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35969|2024-08-03T09:51:59.714Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35969</link>
      <description>TensorFlow is an open source platform for machine learning. The implementation of `Conv2DBackpropInput` requires `input_sizes` to be 4-dimensional. Otherwise, it gives a `CHECK` failure which can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 50156d547b9a1da0144d7babe665cf690305b33c. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35969|2024-08-03T09:51:59.714Z</guid>
      <pubDate>Fri, 16 Sep 2022 20:45:19 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35973|2024-08-03T09:51:59.641Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35973</link>
      <description>TensorFlow is an open source platform for machine learning. If `QuantizedMatMul` is given nonscalar input for: `min_a`, `max_a`, `min_b`, or `max_b` It gives a segfault that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit aca766ac7693bf29ed0df55ad6bfcc78f35e7f48. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35973|2024-08-03T09:51:59.641Z</guid>
      <pubDate>Fri, 16 Sep 2022 21:00:14 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23595|2024-08-03T03:43:46.868Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23595</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. When building an XLA compilation cache, if default settings are used, TensorFlow triggers a null pointer dereference. In the default scenario, all devices are allowed, so `flr-&gt;config_proto` is `nullptr`. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23595|2024-08-03T03:43:46.868Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:13 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35981|2024-08-03T09:51:59.677Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35981</link>
      <description>TensorFlow is an open source platform for machine learning. `FractionalMaxPoolGrad` validates its inputs with `CHECK` failures instead of with returning errors. If it gets incorrectly sized inputs, the `CHECK` failure can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 8741e57d163a079db05a7107a7609af70931def4. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35981|2024-08-03T09:51:59.677Z</guid>
      <pubDate>Fri, 16 Sep 2022 21:15:12 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36018|2024-08-03T09:51:59.752Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36018</link>
      <description>TensorFlow is an open source platform for machine learning. If `RaggedTensorToVariant` is given a `rt_nested_splits` list that contains tensors of ranks other than one, it results in a `CHECK` fail that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 88f93dfe691563baa4ae1e80ccde2d5c7a143821. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36018|2024-08-03T09:51:59.752Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:05:15 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35987|2024-08-03T09:51:59.874Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35987</link>
      <description>TensorFlow is an open source platform for machine learning. `DenseBincount` assumes its input tensor `weights` to either have the same shape as its input tensor `input` or to be length-0. A different `weights` shape will trigger a `CHECK` fail that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit bf4c14353c2328636a18bfad1e151052c81d5f43. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35987|2024-08-03T09:51:59.874Z</guid>
      <pubDate>Fri, 16 Sep 2022 21:40:24 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35965|2024-08-03T09:51:59.513Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35965</link>
      <description>TensorFlow is an open source platform for machine learning. If `LowerBound` or `UpperBound` is given an empty`sorted_inputs` input, it results in a `nullptr` dereference, leading to a segfault that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit bce3717eaef4f769019fd18e990464ca4a2efeea. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35965|2024-08-03T09:51:59.513Z</guid>
      <pubDate>Fri, 16 Sep 2022 20:25:09 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35972|2024-08-03T09:51:59.305Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35972</link>
      <description>TensorFlow is an open source platform for machine learning. If `QuantizedBiasAdd` is given `min_input`, `max_input`, `min_bias`, `max_bias` tensors of a nonzero rank, it results in a segfault that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 785d67a78a1d533759fcd2f5e8d6ef778de849e0. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35972|2024-08-03T09:51:59.305Z</guid>
      <pubDate>Fri, 16 Sep 2022 21:00:19 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23593|2024-08-03T03:43:46.906Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23593</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. The `simplifyBroadcast` function in the MLIR-TFRT infrastructure in TensorFlow is vulnerable to a segfault (hence, denial of service), if called with scalar shapes. If all shapes are scalar, then `maxRank` is 0, so we build an empty `SmallVector`. The fix will be included in TensorFlow 2.8.0. This is the only affected version.</description>
      <guid isPermaLink="false">CVE-2022-23593|2024-08-03T03:43:46.906Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:08 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23566|2024-08-03T03:43:46.603Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23566</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. TensorFlow is vulnerable to a heap OOB write in `Grappler`. The `set_output` function writes to an array at the specified index. Hence, this gives a malicious user a write primitive. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23566|2024-08-03T03:43:46.603Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:33 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29202|2024-08-03T06:17:54.103Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29202</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.ragged.constant` does not fully validate the input arguments. This results in a denial of service by consuming all available memory. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29202|2024-08-03T06:17:54.103Z</guid>
      <pubDate>Fri, 20 May 2022 22:55:13 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29198|2024-08-03T06:17:54.186Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29198</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.SparseTensorToCSRSparseMatrix` does not fully validate the input arguments. This results in a `CHECK`-failure which can be used to trigger a denial of service attack. The code assumes `dense_shape` is a vector and `indices` is a matrix (as part of requirements for sparse tensors) but there is no validation for this. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29198|2024-08-03T06:17:54.186Z</guid>
      <pubDate>Fri, 20 May 2022 21:50:12 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35982|2024-08-03T09:51:59.337Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35982</link>
      <description>TensorFlow is an open source platform for machine learning. If `SparseBincount` is given inputs for `indices`, `values`, and `dense_shape` that do not make a valid sparse tensor, it results in a segfault that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 40adbe4dd15b582b0210dfbf40c243a62f5119fa. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35982|2024-08-03T09:51:59.337Z</guid>
      <pubDate>Fri, 16 Sep 2022 21:30:13 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23559|2024-08-03T03:43:46.566Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23559</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. An attacker can craft a TFLite model that would cause an integer overflow in embedding lookup operations. Both `embedding_size` and `lookup_size` are products of values provided by the user. Hence, a malicious user could trigger overflows in the multiplication. In certain scenarios, this can then result in heap OOB read/write. Users are advised to upgrade to a patched version.</description>
      <guid isPermaLink="false">CVE-2022-23559|2024-08-03T03:43:46.566Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:37 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29205|2024-08-03T06:17:54.256Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29205</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, there is a potential for segfault / denial of service in TensorFlow by calling `tf.compat.v1.*` ops which don't yet have support for quantized types, which was added after migration to TensorFlow 2.x. In these scenarios, since the kernel is missing, a `nullptr` value is passed to `ParseDimensionValue` for the `py_value` argument. Then, this is dereferenced, resulting in segfault. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29205|2024-08-03T06:17:54.256Z</guid>
      <pubDate>Fri, 20 May 2022 22:25:14 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23558|2024-08-03T03:43:46.558Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23558</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. An attacker can craft a TFLite model that would cause an integer overflow in `TfLiteIntArrayCreate`. The `TfLiteIntArrayGetSizeInBytes` returns an `int` instead of a `size_t. An attacker can control model inputs such that `computed_size` overflows the size of `int` datatype. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23558|2024-08-03T03:43:46.558Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:43 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36027|2024-08-03T09:51:59.867Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36027</link>
      <description>TensorFlow is an open source platform for machine learning. When converting transposed convolutions using per-channel weight quantization the converter segfaults and crashes the Python process. We have patched the issue in GitHub commit aa0b852a4588cea4d36b74feb05d93055540b450. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36027|2024-08-03T09:51:59.867Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:50:16 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29211|2024-08-03T06:17:54.256Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29211</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.histogram_fixed_width` is vulnerable to a crash when the values array contain `Not a Number` (`NaN`) elements. The implementation assumes that all floating point operations are defined and then converts a floating point result to an integer index. If `values` contains `NaN` then the result of the division is still `NaN` and the cast to `int32` would result in a crash. This only occurs on the CPU implementation. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29211|2024-08-03T06:17:54.256Z</guid>
      <pubDate>Fri, 20 May 2022 23:20:15 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23594|2024-08-03T03:43:46.930Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23594</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. The TFG dialect of TensorFlow (MLIR) makes several assumptions about the incoming `GraphDef` before converting it to the MLIR-based dialect. If an attacker changes the `SavedModel` format on disk to invalidate these assumptions and the `GraphDef` is then converted to MLIR-based IR then they can cause a crash in the Python interpreter. Under certain scenarios, heap OOB read/writes are possible. These issues have been discovered via fuzzing and it is possible that more weaknesses exist. We will patch them as they are discovered.</description>
      <guid isPermaLink="false">CVE-2022-23594|2024-08-03T03:43:46.930Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:11 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35993|2024-08-03T09:51:59.788Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35993</link>
      <description>TensorFlow is an open source platform for machine learning. When `SetSize` receives an input `set_shape` that is not a 1D tensor, it gives a `CHECK` fails that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit cf70b79d2662c0d3c6af74583641e345fc939467. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35993|2024-08-03T09:51:59.788Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:20:25 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23574|2024-08-03T03:43:46.946Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23574</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. There is a typo in TensorFlow's `SpecializeType` which results in heap OOB read/write. Due to a typo, `arg` is initialized to the `i`th mutable argument in a loop where the loop index is `j`. Hence it is possible to assign to `arg` from outside the vector of arguments. Since this is a mutable proto value, it allows both read and write to outside of bounds data. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, and TensorFlow 2.6.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23574|2024-08-03T03:43:46.946Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:35 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23586|2024-08-03T03:43:46.905Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23586</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. A malicious user can cause a denial of service by altering a `SavedModel` such that assertions in `function.cc` would be falsified and crash the Python interpreter. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23586|2024-08-03T03:43:46.905Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:19 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29204|2024-08-03T06:17:54.286Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29204</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.UnsortedSegmentJoin` does not fully validate the input arguments. This results in a `CHECK`-failure which can be used to trigger a denial of service attack. The code assumes `num_segments` is a positive scalar but there is no validation. Since this value is used to allocate the output tensor, a negative value would result in a `CHECK`-failure (assertion failure), as per TFSA-2021-198. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29204|2024-08-03T06:17:54.286Z</guid>
      <pubDate>Fri, 20 May 2022 22:40:13 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23564|2024-08-03T03:43:46.543Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23564</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. When decoding a resource handle tensor from protobuf, a TensorFlow process can encounter cases where a `CHECK` assertion is invalidated based on user controlled arguments. This allows attackers to cause denial of services in TensorFlow processes. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23564|2024-08-03T03:43:46.543Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:41 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35985|2024-08-03T09:51:59.752Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35985</link>
      <description>TensorFlow is an open source platform for machine learning. If `LRNGrad` is given an `output_image` input tensor that is not 4-D, it results in a `CHECK` fail that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit bd90b3efab4ec958b228cd7cfe9125be1c0cf255. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35985|2024-08-03T09:51:59.752Z</guid>
      <pubDate>Fri, 16 Sep 2022 21:40:20 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36019|2024-08-03T09:51:59.760Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36019</link>
      <description>TensorFlow is an open source platform for machine learning. If `FakeQuantWithMinMaxVarsPerChannel` is given `min` or `max` tensors of a rank other than one, it results in a `CHECK` fail that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 785d67a78a1d533759fcd2f5e8d6ef778de849e0. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36019|2024-08-03T09:51:59.760Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:05:10 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35964|2024-08-03T09:51:59.312Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35964</link>
      <description>TensorFlow is an open source platform for machine learning. The implementation of `BlockLSTMGradV2` does not fully validate its inputs. This results in a a segfault that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 2a458fc4866505be27c62f81474ecb2b870498fa. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35964|2024-08-03T09:51:59.312Z</guid>
      <pubDate>Fri, 16 Sep 2022 20:25:14 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23571|2024-08-03T03:43:46.554Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23571</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. When decoding a tensor from protobuf, a TensorFlow process can encounter cases where a `CHECK` assertion is invalidated based on user controlled arguments, if the tensors have an invalid `dtype` and 0 elements or an invalid shape. This allows attackers to cause denial of services in TensorFlow processes. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23571|2024-08-03T03:43:46.554Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:34 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23573|2024-08-03T03:43:46.500Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23573</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. The implementation of `AssignOp` can result in copying uninitialized data to a new tensor. This later results in undefined behavior. The implementation has a check that the left hand side of the assignment is initialized (to minimize number of allocations), but does not check that the right hand side is also initialized. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23573|2024-08-03T03:43:46.500Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:28 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36012|2024-08-03T09:51:59.629Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36012</link>
      <description>TensorFlow is an open source platform for machine learning. When `mlir::tfg::ConvertGenericFunctionToFunctionDef` is given empty function attributes, it crashes. We have patched the issue in GitHub commit ad069af92392efee1418c48ff561fd3070a03d7b. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36012|2024-08-03T09:51:59.629Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:55:15 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35995|2024-08-03T09:51:59.521Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35995</link>
      <description>TensorFlow is an open source platform for machine learning. When `AudioSummaryV2` receives an input `sample_rate` with more than one element, it gives a `CHECK` fails that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit bf6b45244992e2ee543c258e519489659c99fb7f. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35995|2024-08-03T09:51:59.521Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:15:12 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23570|2024-08-03T03:43:46.571Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23570</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. When decoding a tensor from protobuf, TensorFlow might do a null-dereference if attributes of some mutable arguments to some operations are missing from the proto. This is guarded by a `DCHECK`. However, `DCHECK` is a no-op in production builds and an assertion failure in debug builds. In the first case execution proceeds to the dereferencing of the null pointer, whereas in the second case it results in a crash due to the assertion failure. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, and TensorFlow 2.6.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23570|2024-08-03T03:43:46.571Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:42 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35935|2024-08-03T09:51:59.075Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35935</link>
      <description>TensorFlow is an open source platform for machine learning. The implementation of SobolSampleOp is vulnerable to a denial of service via CHECK-failure (assertion failure) caused by assuming `input(0)`, `input(1)`, and `input(2)` to be scalar. This issue has been patched in GitHub commit c65c67f88ad770662e8f191269a907bf2b94b1bf. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35935|2024-08-03T09:51:59.075Z</guid>
      <pubDate>Fri, 16 Sep 2022 19:35:15 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23591|2024-08-03T03:43:46.959Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23591</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. The `GraphDef` format in TensorFlow does not allow self recursive functions. The runtime assumes that this invariant is satisfied. However, a `GraphDef` containing a fragment such as the following can be consumed when loading a `SavedModel`. This would result in a stack overflow during execution as resolving each `NodeDef` means resolving the function itself and its nodes. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23591|2024-08-03T03:43:46.959Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:09 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23588|2024-08-03T03:43:46.905Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23588</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. A malicious user can cause a denial of service by altering a `SavedModel` such that Grappler optimizer would attempt to build a tensor using a reference `dtype`. This would result in a crash due to a `CHECK`-fail in the `Tensor` constructor as reference types are not allowed. The fix will be included in TensorFlow 2.8.0. We will also cherrypick this commit on TensorFlow 2.7.1, TensorFlow 2.6.3, and TensorFlow 2.5.3, as these are also affected and still in supported range.</description>
      <guid isPermaLink="false">CVE-2022-23588|2024-08-03T03:43:46.905Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:21 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29203|2024-08-03T06:17:54.162Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29203</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.SpaceToBatchND` (in all backends such as XLA and handwritten kernels) is vulnerable to an integer overflow: The result of this integer overflow is used to allocate the output tensor, hence we get a denial of service via a `CHECK`-failure (assertion failure), as in TFSA-2021-198. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29203|2024-08-03T06:17:54.162Z</guid>
      <pubDate>Fri, 20 May 2022 22:50:11 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-29194|2024-08-03T06:17:54.237Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-29194</link>
      <description>TensorFlow is an open source platform for machine learning. Prior to versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4, the implementation of `tf.raw_ops.DeleteSessionTensor` does not fully validate the input arguments. This results in a `CHECK`-failure which can be used to trigger a denial of service attack. Versions 2.9.0, 2.8.1, 2.7.2, and 2.6.4 contain a patch for this issue.</description>
      <guid isPermaLink="false">CVE-2022-29194|2024-08-03T06:17:54.237Z</guid>
      <pubDate>Fri, 20 May 2022 21:10:12 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-23592|2024-08-03T03:43:46.895Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-23592</link>
      <description>Tensorflow is an Open Source Machine Learning Framework. TensorFlow's type inference can cause a heap out of bounds read as the bounds checking is done in a `DCHECK` (which is a no-op during production). An attacker can control the `input_idx` variable such that `ix` would be larger than the number of values in `node_t.args`. The fix will be included in TensorFlow 2.8.0. This is the only affected version.</description>
      <guid isPermaLink="false">CVE-2022-23592|2024-08-03T03:43:46.895Z</guid>
      <pubDate>Fri, 04 Feb 2022 22:32:14 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-35967|2024-08-03T09:51:59.308Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-35967</link>
      <description>TensorFlow is an open source platform for machine learning. If `QuantizedAdd` is given `min_input` or `max_input` tensors of a nonzero rank, it results in a segfault that can be used to trigger a denial of service attack. We have patched the issue in GitHub commit 49b3824d83af706df0ad07e4e677d88659756d89. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-35967|2024-08-03T09:51:59.308Z</guid>
      <pubDate>Fri, 16 Sep 2022 20:35:10 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2022-36000|2024-08-03T09:51:59.697Z -- tensorflow -- tensorflow
</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2022-36000</link>
      <description>TensorFlow is an open source platform for machine learning. When `mlir::tfg::ConvertGenericFunctionToFunctionDef` is given empty function attributes, it gives a null dereference. We have patched the issue in GitHub commit aed36912609fc07229b4d0a7b44f3f48efc00fd0. The fix will be included in TensorFlow 2.10.0. We will also cherrypick this commit on TensorFlow 2.9.1, TensorFlow 2.8.1, and TensorFlow 2.7.2, as these are also affected and still in supported range. There are no known workarounds for this issue.</description>
      <guid isPermaLink="false">CVE-2022-36000|2024-08-03T09:51:59.697Z</guid>
      <pubDate>Fri, 16 Sep 2022 22:30:25 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2024-3660|2024-08-01T20:20:00.692Z -- tensorflow -- keras</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2024-3660</link>
      <description>A arbitrary code injection vulnerability in TensorFlow's Keras framework (&lt;2.13) allows attackers to execute arbitrary code with the same permissions as the application using a model that allow arbitrary code irrespective of the application.</description>
      <guid isPermaLink="false">CVE-2024-3660|2024-08-01T20:20:00.692Z</guid>
      <pubDate>Tue, 16 Apr 2024 20:09:26 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2023-33976|2024-08-01T20:14:02.779Z -- tensorflow -- tensorflow</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2023-33976</link>
      <description>TensorFlow is an end-to-end open source platform for machine learning. `array_ops.upper_bound` causes a segfault when not given a rank 2 tensor. The fix will be included in TensorFlow 2.13 and will also cherrypick this commit on TensorFlow 2.12.</description>
      <guid isPermaLink="false">CVE-2023-33976|2024-08-01T20:14:02.779Z</guid>
      <pubDate>Tue, 30 Jul 2024 19:27:14 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2024-3660|2024-07-29T19:38:15.693Z -- tensorflow -- keras</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2024-3660</link>
      <description>A arbitrary code injection vulnerability in TensorFlow's Keras framework (&lt;2.13) allows attackers to execute arbitrary code with the same permissions as the application using a model that allow arbitrary code irrespective of the application.</description>
      <guid isPermaLink="false">CVE-2024-3660|2024-07-29T19:38:15.693Z</guid>
      <pubDate>Tue, 16 Apr 2024 20:09:26 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2024-3660|2024-04-16T20:09:26.220Z -- tensorflow -- keras</title>
      <link>https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2024-3660</link>
      <description>A arbitrary code injection vulnerability in TensorFlow's Keras framework (&lt;2.13) allows attackers to execute arbitrary code with the same permissions as the application using a model that allow arbitrary code irrespective of the application.</description>
      <guid isPermaLink="false">CVE-2024-3660|2024-04-16T20:09:26.220Z</guid>
      <pubDate>Tue, 16 Apr 2024 20:09:26 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2023-33976|2024-07-30T19:27:14.532Z -- tensorflow -- tensorflow</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2023-33976</link>
      <description>TensorFlow is an end-to-end open source platform for machine learning. `array_ops.upper_bound` causes a segfault when not given a rank 2 tensor. The fix will be included in TensorFlow 2.13 and will also cherrypick this commit on TensorFlow 2.12.</description>
      <guid isPermaLink="false">CVE-2023-33976|2024-07-30T19:27:14.532Z</guid>
      <pubDate>Tue, 30 Jul 2024 19:27:14 +0000</pubDate>
    </item>
    <item>
      <title>CVE-2023-33976|2024-08-02T15:54:14.208Z -- tensorflow -- tensorflow</title>
      <link>https://www.cve.org/CVERecord?id=CVE-2023-33976</link>
      <description>TensorFlow is an end-to-end open source platform for machine learning. `array_ops.upper_bound` causes a segfault when not given a rank 2 tensor. The fix will be included in TensorFlow 2.13 and will also cherrypick this commit on TensorFlow 2.12.</description>
      <guid isPermaLink="false">CVE-2023-33976|2024-08-02T15:54:14.208Z</guid>
      <pubDate>Tue, 30 Jul 2024 19:27:14 +0000</pubDate>
    </item>
  </channel>
</rss>
